hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/FrameUtils.java

```diff
@@ -257,6 +260,32 @@ public static boolean unqualifiedDelegateDetected(final MessageFrame frame) {
         return true;
     }
 
+    /**
+     * Returns true if the given frame is a call to a contract that must be present based on feature flag settings.
+     * @param frame
+     * @param address to check for possible grandfathering
+     * @param featureFlags
+     * @return
+     */
+    public static boolean contractRequired(
```

Comment:
Q: `contractMustBePresent` and `contractRequired` seem similar in purpose and naming but are in different classes.
Are they essentially overloads with one taking a `ContractId` and the other taking an `address` and should they be collocated?

Reply:
`contractMustBePresent` is private to FrameBuilder.  I needed this method to construct a `messageFrame` which is then used everywhere else because it is available when running the evm.